### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by GurevichVarvara**
For that time I also have some questions.

First question: when user adds item (todo, for example) I create item-div with all necessary html-tags inside function in js-file. I found that that's a way to perform SPA, but it's not very convinient cause when I faced with the problem of changing html-structure of item-div I had to make changes both in html and js files. So is that the only way of creating SPA logic, or maybe it would be a greater idea to have invisible item-div element inside html-document and when js would have to create one more item-div, it would copy that item-div from html-document and then would change properties and append it to DOM? 
Never the less I don't think the way I organized SPA logic is wrong, or you don't think the same as me?

Second question preview: I have trash page and form where user could change types of items and sorting order. I have only two categories of items (todo and habit) but you asked me to add "all"-category for the future, and that particular category raises the correlations ("todos" + "all" = "todos" + "habits" = "habits" + "all" = "all"). To check if the categories of items user had chosen equal to categories of items that are already displaying, I use global variable in the trash_page.js file in which I keep options of list which is displaying. I do it for preventing redundant comunication with server. 
Second question: Should I use something else insted of verible for that?

That's all. I changed register-rules and cleaned-up my db, so could you please re-regester if you've done it before.

Link remains the same: http://62.171.133.174:8000/

### CODE REVIEW

**Comment by GurevichVarvara** [comment]
I forgot to mention, I don’t like how mobile version of todos-page looks if I display only these section of todos which is filled by items, because if all sections are in front of your eyes even if they're empty, you  may determine which categories are filled without reading the title cause you would have a stracture in your mind.

**Comment by gmltA** [comment]
Regarding first question: you could've used `template` tag to instantiate new page elements. That's exactly what you described in your proposal)

Regarding second one: you can change 'All' to be a button, that selects all other checkboxes.
It's not a good strategy to store this value as global. Use checkboxes state as source of truth instead.



> I forgot to mention, I don’t like how mobile version of todos-page looks if I display only these section of todos which is filled by items or when I put empty sections down, because if all sections are in front of your eyes even if they're empty in particular sequence, you may determine which categories you look at without reading the title cause you would have a stracture in your mind.

Well, it's a matter personal habit) Just do it in a way that will bother you less.

File: TODO/static/CSS/todo_styles.css [hunk 1]
```diff
@@ -29,12 +29,12 @@ main {
 }
 
 .todo-item {
-	background: rgba(221, 76, 14);
+	background: rgba(221, 76, 14, 1);
```
Comment (gmltA) [inline TODO/static/CSS/todo_styles.css:32]:
```suggestion
	background: rgb(221, 76, 14);
```

File: TODO/static/JS/todo_habit_adding.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
+window.onload=function() {
```
Comment (gmltA) [inline TODO/static/JS/todo_habit_adding.js:1]: It is better to use `DOMContentLoaded` event

**Comment by GurevichVarvara** [comment]
> Will there be SPA at the end?)


Damn... Now I see that it is about whole aplication and not only single page. But sure, SPA would be by tomorrow

**Comment by gmltA** [comment]
Great. Looking forward to it!

**Comment by GurevichVarvara** [comment]
Hmm, could I put all main-divs into one page and switch between them? It sounds stupid, but it the easiest solution that came to my mind)

**Comment by gmltA** [comment]
> Hmm, could I put all main-divs into one page and switch between them? It sounds stupid, but it the easiest solution that came to my mind)

No, that's not a good solution. The common way here is to put parts of markup in JS strings. I'd advise to apply some encapsulation and split code by components so that markup and component-specific logic will be in one place. You could also make use of [Web Components](https://developer.mozilla.org/en-US/docs/Web/Web_Components), but it will take some time to get used to them.

**Comment by GurevichVarvara** [comment]
I did it!

**Comment by GurevichVarvara** [comment]
I had a look at the example you attached and I'm not sure about the idea to add '#' before main-part of a url-path, but it works and works well. What do you think about that?

File: TODO/static/JS/login-register_form.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+
+function are_pass_and_confirmation_equal(password, confirmation) {
+    let result = false;
+
+    if (password === confirmation) {
+        result = true;
+    }
+
+    return result;
```
Comment (gmltA) [inline TODO/static/JS/login-register_form.js:73]:
```suggestion
    return password === confirmation;
```

**Comment by gmltA** [comment]
> I'm not sure about the idea to add '#' before main-part of a url-path

The idea is to handle window location change via `hashchange` event, because initially developers had no possibility to intercept history modification events.

**Comment by GurevichVarvara** [comment]
> Right. So what's left?


That’s all I thought. Functional fully completed. Or maybe you want me to add something?

**Comment by GurevichVarvara** [comment]
Anything else?

**Pull request merged**